### PR TITLE
youtube integrations part 1

### PR DESCRIPTION
**Pull request by greatsamist**
✨ This PR introduces the first part of a new feature that allows users to upload videos to YouTube directly from our platform. It includes an upload button, social models for organizations, and a comprehensive flow for adding and managing linked accounts.

Impact
This feature significantly enhances our platform by enabling seamless video uploads to YouTube, improving user engagement and providing a more integrated experience. The new social models for organizations ensure better management and flexibility for linked social media accounts.

User Impact
Users can now upload videos to YouTube through a straightforward interface. When they click the upload button, a modal prompts them to select from their linked accounts or connect a new account. This flow also checks if the user has a YouTube account and handles the process accordingly.

image

### CODE REVIEW

File: packages/server/src/utils/youtube.ts [hunk 1]
```diff
@@ -103,7 +98,7 @@ export async function uploadToYouTube(
           defaultAudioLanguage: 'en',
         },
         status: {
-          privacyStatus: event[0].unlisted ? 'unlisted' : 'public',
+          privacyStatus: session.published ? 'public' : 'unlisted',
```
Comment (Mario-SO) [inline packages/server/src/utils/youtube.ts:101]: Why were we previously setting the status based on event and now based on session?

File: packages/app/lib/utils/googleAuth.ts [hunk 2]
```diff
@@ -1,23 +1,98 @@
...
+    oauth2Client.setCredentials({
+      access_token: data.access_token,
+    })
+
+    const people = google.people({
+      version: 'v1',
+      auth: oauth2Client,
+    })
+
+    const peopleResponse = await people.people.get({
```
Comment (Mario-SO) [inline packages/app/lib/utils/googleAuth.ts:35]: people.people?

File: packages/app/app/studio/[organization]/library/[session]/components/UploadToYoutubeButton.tsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    )
+    // Encode the redirect URL
+    const authUrl = `https://accounts.google.com/o/oauth2/auth?client_id=${clientId}&redirect_uri=${redirectUri}&response_type=code&access_type=offline&scope=https://www.googleapis.com/auth/youtube.upload https://www.googleapis.com/auth/youtube.readonly https://www.googleapis.com/auth/youtube https://www.googleapis.com/auth/userinfo.profile https://www.googleapis.com/auth/userinfo.email&state=${state}`
+    window.location.href = authUrl
+  }
+  const hasSocials = organization?.socials?.length
+    ? organization?.socials?.length > 0
+    : false
+
+  const handlePublish = () => {}
```
Comment (Mario-SO) [inline packages/app/app/studio/[organization]/library/[session]/components/UploadToYoutubeButton.tsx:75]: I guess this is for part two of the PR right?

Comment (greatsamist) [reply packages/app/app/studio/[organization]/library/[session]/components/UploadToYoutubeButton.tsx:75]: Yes, awaiting @Eric-Vondee feedback on the best form of payload to send from the frontend to the publish api

File: packages/app/lib/utils/googleAuth.ts (hunk 2 above)
Comment (greatsamist) [reply packages/app/lib/utils/googleAuth.ts:35]: full scope
```
 const people = google.people({
      version: 'v1',
      auth: oauth2Client,
    })

    const peopleResponse = await people.people.get({
      resourceName: 'people/me',
      personFields: 'names,photos,emailAddresses',
    })
    ```
    resources

**Review by Mario-SO — APPROVED** [review]

**Pull request merged**